datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -462,6 +462,7 @@ fn date_bin_months_interval(stride_months: i64, source: i64, origin: i64) -> i64
     bin_time.timestamp_nanos()
 }
 
+#[allow(deprecated)]
```

Comment:
```
   --> datafusion/physical-expr/src/datetime_expressions.rs:469:22
    |
469 |     DateTime::<Utc>::from_utc(date, Utc)
    |                      ^^^^^^^^
    |
    = note: `-D deprecated` implied by `-D warnings`
```

I think this is not from substrait upgrade, but CI complains about it.

Reply:
https://github.com/apache/arrow-datafusion/pull/7451 tries to fix this.

Reply:
Ah, okay, I will rebase after the fix is merged.